### ISSUE TITLE
Extend FieldData to have different validation statuses

### ISSUE THREAD

**Issue by nbhargava** [ant-design/ant-design#36601]
### What problem does this feature solve?

When the `form.setFields` API is invoked, you can only set different types of errors, despite the fact that validation type can include `warning`. Extending the API for FieldData would let us programmatically set warnings in addition to errors.

### What does the proposed API look like?

```
interface FieldDataError {
  text: string;
  validateStatus: 'success' | 'warning' | 'error' | 'validating';
}

interface FieldData {
  name: string | number | (string | number)[];
  value?: any;
  touched?: boolean;
  validating?: boolean;
  errors?: FieldDataError[];
}
```

**Comment by afc163** [comment]
You can set `validateStatus="warning"` on Form.Item.

**Comment by nbhargava** [comment]
Yes, but not via the setField API! Given that there is a way to imperatively set the errors, it would be great if you could set a warning via the same imperative function call.

**Comment by afc163** [comment]
In antd's Form, `validateStatus` is designed to be an UI presentation status but field state.
